src/main/scala/pureconfig/ConfigFieldMapping.scala

```diff
@@ -0,0 +1,24 @@
+package pureconfig
+
+trait ConfigFieldMapping[T] {
```

Comment:
@jcazevedo sorry to keep you wait for this PR. The `String => String` was my first idea for this feature and I think I like it because it's very simple. I'm not sure what do you mean with the  `ConfigFieldMapping` and how you can configure for specific types. What kind of Map do you have in mind?